ppb_vector/vector2.py

```diff
@@ -154,10 +164,8 @@ def __new__(cls, *args, **kwargs):
         except ValueError:
             raise TypeError(f"{type(y).__name__} object not convertable to float")
 
-        return self
-
     def __reduce__(self):
-        return type(self).__new__, (type(self), self.x, self.y)
+        return type(self), (self.x, self.y)
```

Comment:
Using `type(self).__new__` there violates the protocol for [object construction]:

> If `__new__` returns an instance of `cls`, then the new instance’s `__init__()` method will be invoked like `__init__(self[, ...])`, where `self` is the new instance and the remaining arguments are the same as were passed to `__new__`.

[object construction]: https://docs.python.org/3/reference/datamodel.html#object.__new__

Reply:
Extracted to #157.